### PR TITLE
World Saves

### PR DESCRIPTION
**Pull request by Z1Meme**

### CODE REVIEW

**Comment by Z1Meme** [comment]
@QuentinGruber Character saving works for both single and multiplayer (should not require a database reset), and vehicle saving is ready for when we have positionupdates fixed since we have some cleanup to do there and they don't work correctly rn.

**Comment by Z1Meme** [comment]
Branch can be deleted btw

**Comment by QuentinGruber** [comment]
I'm reviewing that today @Z1Meme 🤠

**Comment by QuentinGruber** [comment]
Also this should be a feature easy to disable ( some servers don't want to store characters data and others )

**Comment by QuentinGruber** [comment]
So i've spot 3 issues:

* Old characters can't be used
* Death from collision damage on spawn happens almost everytime
* I saw the log once and it only concern solo play but we shouldn't save circular structure to json ( didn't know which structure it was )

For the code and the way you structured it, it's well done 🤠

File: src/servers/ZoneServer2016/zonepackethandlers.ts [hunk 1]
```diff
@@ -814,7 +814,7 @@ export class zonePacketHandlers {
         server.setGodMode(client, false);
         client.character.tempGodMode = false;
       }
-      client.character.positionUpdate = packet.data;
+      //client.character.positionUpdate = packet.data;
```
Comment (QuentinGruber) [inline src/servers/ZoneServer2016/zonepackethandlers.ts:817]: if we remove it, we should remove the usage of it when sending LightweightToFullPc

**Review by QuentinGruber — CHANGES_REQUESTED** [review]

**Comment by QuentinGruber** [comment]
> So i've spot 3 issues:
> 
> * Old characters can't be used
> * Death from collision damage on spawn happens almost everytime
> * I saw the log once and it only concern solo play but we shouldn't save circular structure to json ( didn't know which structure it was )
> 
> For the code and the way you structured it, it's well done 🤠

Also 2015 version is broken with these changes, maybe we should create legacy functions for it

**Comment by Z1Meme** [comment]
> > So i've spot 3 issues:
> > 
> > * Old characters can't be used
> > * Death from collision damage on spawn happens almost everytime
> > * I saw the log once and it only concern solo play but we shouldn't save circular structure to json ( didn't know which structure it was )
> > 
> > For the code and the way you structured it, it's well done 🤠
> 
> Also 2015 version is broken with these changes, maybe we should create legacy functions for it

- Old characters should just be deleted, they don't really matter anyways
- Death / damage from collision damage is fixed
- Can't reproduce that circular json issue
- Can't reproduce the 2015 issue, it works for me on both singleplayer and multiplayer, and as far as I know nothing was edited having to do with 2015

**Comment by Z1Meme** [comment]
> Also this should be a feature easy to disable ( some servers don't want to store characters data and others )

Done
`process.env.ENABLE_SAVES`

**Comment by Z1Meme** [comment]
@QuentinGruber this is ready to merge when you have some time

**Comment by QuentinGruber** [comment]
I was thinking we could add some versioning to this. In the Worlds collection each world should have the version of the save system it generated, so that it wipes the world automatically if a newer version is used

**Review by QuentinGruber — APPROVED** [review]
🥳

**Pull request merged**